### DISCUSSION TITLE
Refresh nested Livewire component after delete()

### DISCUSSION THREAD

**Discussion by maxwoerner** [livewire/livewire#3197] [1. Help] [answered]
I have a nested Livewire component for Groups. Each Group contains multiple Guests.

```blade
@foreach ($event->groups as $group)

@livewire('guest.create', ['group' => $group], key(time().$group->id))

@endforeach
```

When I delete one of the Guests, I want to refresh the component so the changes are shown.

```php
class ManageGuests extends Component
{

    public $group;

    public $guests;

    protected $listeners = [
        'refresh' => '$refresh'
    ];

    public function mount()
    {
        $this->guests = $this->group->guests;
    }

    public function render()
    {
        return view('livewire.guest.create');
    }

    public function delete($id)
    {
        $guest = Guest::find($id);

        $guest->delete();

        $this->emit('refresh');

    }
}
```

However, I always get below error:
     _**'ErrorException Undefined array key 0'**_

I am pretty sure this is a Dom Diffing Issue somehow. However, I am not able to fix it... Any help would be highly appreciated. Thank you!

**Comment by drbornot** [comment]
How do you retrieve this value $event->groups??? if you make it dd...what you get?

**Reply by maxwoerner** [reply]
$event is passed as a parameter in the blade directive:

`@livewire('group.manage', ['event' => $event])`

It has a hasMany() relation with Groups:

```
class Event extends Model
{
    /**
     * Get the groups for the event.
     */
    public function groups()
    {
        return $this->hasMany(Group::class);
    }
}
```

It's basically a three step relation: Event has many Group has many Guest. Hope this clarifies it.

**Comment by joshhanley** [comment, marked as answer]
@mxwebdev you need to "reload" the collection `$guests` after you delete.

So in your delete method at the end before the emit, put this
```php
$this->group->load('guests'); // This line gets all the guests fresh from the db

$this->guests = $this->group->guests; // Then reassign back to the component property.
```

Hope this helps!

**Reply by maxwoerner** [reply]
Works like a charm. You saved my day. Thank you very much!

**Reply by joshhanley** [reply]
No worries 🙂
